### PR TITLE
[ci-visibility] [do not merge] CI Visibility Changes for Ruby `1.0.0`

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?
Update the instructions for the ruby tracer when `1.0.0` is stable. 

### Motivation
Keep up to date with ruby tracer version.

### Preview
https://docs-staging.datadoghq.com/juan-fernandez/fix-breaking-changes-ruby/continuous_integration/setup_tests/ruby/?tab=cucumber

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/setup_tests/ruby.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ To activate your integration, add the following code to your application:
 require 'cucumber'
 require 'datadog/ci'
 
-Datadog.configure do |c|
+Datadog::CI.configure do |c|
   # Only activates test instrumentation on CI
   c.tracer.enabled = (ENV["DD_ENV"] == "ci")
```
Comment (ivoanjo) [inline content/en/continuous_integration/setup_tests/ruby.md:65]: I don't think this is correct.

The new configuration model separates global config, and per-product config. Each can only access its own configuration.

So looking at the validation code:

```ruby
require 'datadog/core/configuration/validation_proxy'

module Datadog
  module Tracing
    module Configuration
      # Forwards tracing configuration settings
      class ValidationProxy < Core::Configuration::ValidationProxy
        FORWARDED_METHODS = [
          :analytics,
          :distributed_tracing,
          :instrument,
          :instrumented_integrations,
          :log_injection,
          :log_injection=,
          :reduce_log_verbosity,
          :report_hostname,
          :report_hostname=,
          :sampling,
          :test_mode,
          :tracer
        ].freeze

        def_delegators \
          :settings,
          *FORWARDED_METHODS
      end
    end
  end
end
require 'forwardable'

module Datadog
  module Core
    module Configuration
      # Forwards configuration settings that are permitted,
      # but raises errors for access to anything else.
      class ValidationProxy
        extend Forwardable

        FORWARDED_METHODS = [
          :reset!
        ].freeze

        def_delegators \
          :settings,
          *FORWARDED_METHODS

        def initialize(settings)
          @settings = settings
        end

        protected

        attr_reader :settings

        # Forwards global configuration settings
        class Global < self
          FORWARDED_METHODS = [
            :agent,
            :api_key,
            :api_key=,
            :diagnostics,
            :env,
            :env=,
            :logger,
            :runtime_metrics,
            :service,
            :service_without_fallback,
            :service=,
            :site,
            :site=,
            :tags,
            :tags=,
            :time_now_provider,
            :time_now_provider=,
            :version,
            :version=
          ].freeze

          def_delegators \
            :settings,
            *FORWARDED_METHODS
        end
      end
    end
  end
end
require 'datadog/core/configuration/validation_proxy'

module Datadog
  module CI
    module Configuration
      # Forwards CI configuration settings
      class ValidationProxy < Core::Configuration::ValidationProxy
        FORWARDED_METHODS = [
          :ci_mode
        ].freeze

        def_delegators \
          :settings,
          *FORWARDED_METHODS
      end
    end
  end
end
```

`c.tracer` can only be set via `Datadog::Tracing.configure`; `c.ci_mode` can only be set via `Datadog::CI.configure`; `c.service` is core, and `use` I think is tracing but comes in via a monkey patch which I'm not finding at the moment (sigh 😢 )

Comment (marcotc) [reply content/en/continuous_integration/setup_tests/ruby.md:65]: I think this is what this specific snippet will look like:
```ruby
Datadog.configure do |c|
  # The name of the service or library under test
  c.service = 'my-ruby-app'
end

Datadog::Tracing.configure do |c|
  # Only activates test instrumentation on CI
  c.tracer.enabled = (ENV["DD_ENV"] == "ci")
  # Enables the Cucumber instrumentation
  c.use :cucumber
end

Datadog::CI.configure do |c|
  # Configures the tracer to ensure results delivery
  c.ci_mode.enabled = true
end
```

The `c.use :cucumber` is definitely questionable, and `Datadog::CI.configure` should support it.

But, to be honest, I think this is so much more complex that what we have today :/

Comment (ivoanjo) [reply content/en/continuous_integration/setup_tests/ruby.md:65]: > But, to be honest, I think this is so much more complex that what we have today :/

Yeah, I don't like it either.

**Comment by juan-fernandez** [comment]
closing in favor of https://github.com/DataDog/documentation/pull/13871

**Pull request closed without merging**